test/react_client/js_interop_helpers_test.dart

```diff
@@ -53,12 +53,16 @@ class Foo {
 @JS()
 external dynamic createArray();
 
+/// Like jsifyAndAllowInterop, but casts to Object
+Object jsifyAndAllowInteropAsObject(Object value) => jsifyAndAllowInterop(value) as Object;
+
 main() {
   group('jsifyAndAllowInterop', () {
     test('converts a List', () {
       final list = [1, 2, 3, 4, 5, 6, 7, 8];
-      final array = jsifyAndAllowInterop(list);
+      final array = jsifyAndAllowInteropAsObject(list);
       expect(array is List, isTrue);
+      array as List;
```

Comment:
Whoa - is this new? Cast once and be done for the remaining scope?

Reply:
Yup! There's a bunch of tiny little type promotion improvements like this added alongside null safety; it's so nice!